src/fu-engine.c

```diff
@@ -1613,6 +1614,144 @@ fu_engine_is_running_offline (FuEngine *self)
 #endif
 }
 
+static gboolean
+fu_engine_offline_setup (GError **error)
+{
+	gint rc;
+	g_autofree gchar *filename = NULL;
+	g_autofree gchar *symlink_target = fu_common_get_path (FU_PATH_KIND_LOCALSTATEDIR_PKG);
+	g_autofree gchar *trigger = fu_common_get_path (FU_PATH_KIND_OFFLINE_TRIGGER);
+
+	g_return_val_if_fail (error == NULL || *error == NULL, FALSE);
+
+	/* does already exist */
+	filename = fu_common_realpath (trigger, NULL);
+	if (g_strcmp0 (filename, symlink_target) == 0) {
+		g_debug ("%s already points to %s, skipping creation",
+			 trigger, symlink_target);
+		return TRUE;
+	}
+
+	/* create symlink for the systemd-system-update-generator */
+	rc = symlink (symlink_target, trigger);
+	if (rc < 0) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_INTERNAL,
+			     "Failed to create symlink %s to %s: %s",
+			     trigger,
+			     "/var/lib", strerror (errno));
```

Comment:
There is `g_strerror`, which compared to `strerror` is thread-safe. Not sure it is important though.

Reply:
This line shouldn't hardcode /var/lib, should look up string to use, no?

Reply:
> There is g_strerror

good catch, thanks.

Reply:
> This line shouldn't hardcode /var/lib

I don't think it matters what the destination is, although I guess it makes sense to chose something sensible.

Reply:
Hah, I already did that, the error message was just wrong. Fixed.